### ISSUE TITLE
Dataset Download

### ISSUE THREAD

**Issue by HaminyG** [dxli94/WLASL#1]
Hi sir,
I am really appreciated your work. But there is an issue, how can I download the dataset? I check your download code, each video has a different URL and type. Could you please share your code to download the dataset?

**Comment by dxli94** [comment]
Creating a push-button solution for downloading WLASL is on our backlog. And please keep an eye on our project page https://dxli94.github.io/WLASL/ for updates. We expect it to appear soon.

For the moment, please have a look at the downloading script provided. In particular, 1) for most videos, you can download it through the implemented method effortlessly. 2) For videos from particular sources (e.g. ASL pro), you need to include a header into the URL request. This is also implemented 3) For YouTube videos, I recommend you consider the tool, youtube-dl, for the purpose.

**Comment by dxli94** [comment]
We have now created a script to download non-YouTube videos. Please check it out.

YouTube downloader is under preparation.

**Comment by dxli94** [comment]
We have now included functionalities to download YouTube videos. All in start_kit/video_downloader.py.

The next step would be to provide a preprocessing script.

**Comment by amin07** [comment]
@dxli94, 
Hello, appreciate your great work! Running your downloader script, in total 19196 videos were downloaded. Is this the correct count? The paper says total count should be 21,083. Also some of my downloaded files have random name instead of numeric  video id like below,
![image](https://user-images.githubusercontent.com/15855504/77712984-a6fe5d00-6fab-11ea-9add-1ddcf4ff98a8.png)

**Comment by dxli94** [comment]
> @dxli94,
> Hello, appreciate your great work! Running your downloader script, in total 19196 videos were downloaded. Is this the correct count? The paper says total count should be 21,083. Also some of my downloaded files have random name instead of numeric video id like below,
> ![image](https://user-images.githubusercontent.com/15855504/77712984-a6fe5d00-6fab-11ea-9add-1ddcf4ff98a8.png)

Dear amin07,

It is great you have downloaded the videos. It shows the downloading script is working fine, which is good.

Regarding your question:
- Each non-youtube video should correspond to exactly one video id in JSON file.
- For each YouTube video, it may contain more than one video instance. For example, if you look the entry with video_id 70212:
```
            {
                "bbox": [
                    110,
                    25,
                    274,
                    240
                ],
                "fps": 25,
                "frame_end": 2249,
                "frame_start": 2150,
                "instance_id": 5,
                "signer_id": 121,
                "source": "northtexas",
                "split": "val",
                "url": "https://www.youtube.com/watch?v=hjS0dQDgbjo",
                "variation_id": 0,
                "video_id": "70212"
            }
```
It basically tells you - the #70212 video instance starts from 2249th to 2150th frames in the video with ID hjS0dQDgbjo.mp4 (or .mkv) in some cases.  So you need to extract these frames yourself to create the #70212 video instance.

I am trying to automate this step. But sorry and you may do it yourself for now.

**Comment by amin07** [comment]
oh. great! Thanks a lot for the explanation. I will let you know in case any issue.

**Comment by dxli94** [comment]
> oh. great! Thanks a lot for the explanation. I will let you know in case any issue.

I have now added a script for the purpose mentioned above. By running the preprocess.py, you should be able to get all the videos. 

Please check it out and let me know how it works.

**Comment by amin07** [comment]
It's working great. In total I get 21071 videos. That number seems to be correct. Thanks a lot.

**Comment by dxli94** [comment]
@amin07 Thanks for letting me know. Please see [Section](https://github.com/dxli94/WLASL#requesting-missing-videos) if you would like to acquire missing videos.

**Comment by amin07** [comment]
Hi @dxli94,
Hope you are doing great.
Some of the videos are downloaded damaged. They are not play able and frames cannot be extracted. I was able to figure out 1888 such video samples. Most of them are from the source https://www.asldeafined.com/media/vocabulary/. 
Also, these videos has smaller sizes which suggests they are corrupted,
![image](https://user-images.githubusercontent.com/15855504/79409570-49d34700-7f6c-11ea-9040-fef5ac4ff018.png)
Do you have any idea why those are damaged and how can I achieve good videos?

**Comment by dxli94** [comment]
Hi @amin07,

Thanks for your valuable feedback. It seems they have updated the URLs. 

I've just updated the dead links and please pull the repo and try to re-download the missing ones. 

Thanks.
Dongxu

**Comment by amin07** [comment]
Hello, Thanks for updating. 
i have this other question related to the experiments described in the paper which I could not find an answer to. Did they use variation_id or dialect id as supervision signal for training the models? Also are you planning to upload codes for reproducing the results published?

**Comment by dxli94** [comment]
@amin07 Using dialect info seems an interesting direction. But we did not consider that in the implementation.

We plan to publish the model soon. For urgent use, please temporarily refer to [Google Drive](https://drive.google.com/file/d/1vktQxvRHNS9psOQVKx5-dsERlmiYFRXC/view). Note you may need to preprocess videos and re-train the model on yourself.

**Comment by amin07** [comment]
@dxli94 Thank you

**Comment by sayanlearn** [comment]
Can you specify where exactly we have to install youtube-dl. For some reason, it's skipping only the youtube videos

**Comment by dxli94** [comment]
> Can you specify where exactly we have to install youtube-dl. For some reason, it's skipping only the youtube videos.

Please make sure youtube-dl is installed under any directory of PATH. You can double-check your installation by
```
youtube-dl --version
```

**Comment by amin07** [comment]
> 
> 
> @amin07 Using dialect info seems an interesting direction. But we did not consider that in the implementation.
> 
> We plan to publish the model soon. For urgent use, please temporarily refer to [Google Drive](https://drive.google.com/file/d/1vktQxvRHNS9psOQVKx5-dsERlmiYFRXC/view). Note you may need to preprocess videos and re-train the model on yourself.

hey, looking at your code from the link you provided, it seems to me, that your model was trained on training + validation subset and tested on the test subset. Also, to calculate accuracy you did not take number of individual classes in test split into account. Is the code consistent with the results shown in the paper? Thank you.

**Comment by dxli94** [comment]
> > @amin07 Using dialect info seems an interesting direction. But we did not consider that in the implementation.
> > We plan to publish the model soon. For urgent use, please temporarily refer to [Google Drive](https://drive.google.com/file/d/1vktQxvRHNS9psOQVKx5-dsERlmiYFRXC/view). Note you may need to preprocess videos and re-train the model on yourself.
> 
> hey, looking at your code from the link you provided, it seems to me, that your model was trained on training + validation subset and tested on the test subset. 

We identify the best hyper-params using the validation set. Then we retrain the model combining both training and validation set and report scores on the testing. This is consistent with practices when data in the training set is not sufficient. 

> Also, to calculate accuracy you did not take number of individual classes in test split into account. Is the code consistent with the results shown in the paper? Thank you.

We calculate the so-called "micro-average" accuracy, as there is an obvious imbalance in terms of the number of instances per class. You may adopt other metrics in your own publications.

**Comment by amin07** [comment]
Thank you for the reply.

**Comment by alighofrani95** [comment]
thanks a lot for your interesting work,

unfortunately, I have some issues with downloading the samples on youtube.

by running the video_downloader.py script, It starts to download all samples correctly but even though I have installed the latest version of youtube-dl the youtube samples were skipped and none of them downloaded.

I've tried in both windows and ubuntu...

the log :

gloss: book, video: 69241.
Requesting http://aslbricks.org/New/ASL-Videos/book.mp4
gloss: book, video: 65225.
Requesting https://aslsignbank.haskins.yale.edu/dictionary/protected_media/glossvideo/ASL/BO/BOOK-418.mp4
gloss: book, video: 68011.
Skipping YouTube video 68011
gloss: book, video: 68208.
Skipping YouTube video 68208

**Comment by dxli94** [comment]
Please consider [Request Missing Videos](https://github.com/dxli94/WLASL#requesting-missing-videos) if you encounter issues when downloading the data.

**Comment by hayder218** [comment]
Hello @dxli94  
i just wanted to know how to get the subsets train/test 
should i split it by myself manually ?

**Comment by irg1008** [comment]
> > oh. great! Thanks a lot for the explanation. I will let you know in case any issue.
> 
> I have now added a script for the purpose mentioned above. By running the preprocess.py, you should be able to get all the videos.
> 
> Please check it out and let me know how it works.

Hi, sorry to reply in this issue again, but this is not working for me.

The youtube id name remain after processing the videos.

Thanks for the dataset, it's great!

**Comment by selezenart** [comment]
Hi, issue appeared, after video: 01986. download freezes, and it cannot be skipped

**Comment by dxli94** [comment]
For anyone experiencing the downloading issue, please feel welcome to refer to [README](https://github.com/dxli94/WLASL#requesting-missing--pre-processed-videos) for requesting data.
